samtranslator/model/eventsources/push.py

```diff
@@ -637,6 +639,15 @@ def _add_swagger_integration(self, api, function):
             if method_authorizer or apikey_required_setting is not None:
                 editor.add_auth_to_method(api=api, path=self.Path, method_name=self.Method, auth=self.Auth)
 
+            if self.Auth.get('ResourcePolicy'):
+                resource_policy = self.Auth.get('ResourcePolicy')
+                """
+                At this point the stage name resolution hasnt occured yet so we are going to apply
+                the resource policy on all stages of the API
+                """
+                editor.add_resource_policy(resource_policy=resource_policy,
```

Comment:
`condition` is in scope (see line 595), you should be able to pass the `condition` in to the path and set it on the resource policy inside of an Fn::If statement.

You'll have to test if `AWS::NoValue` works inside of a resource policy, I assume that it will.